ampd/src/cosmos.rs

```diff
@@ -603,4 +721,31 @@ mod tests {
             "res_bytes": res.to_bytes().unwrap()
         }));
     }
+
+    #[test]
+    fn ensure_smart_contract_state_req_res_serialization_do_not_change() {
+        let req = QuerySmartContractStateRequest {
+            address: "axelar1q95p9fntvqn6jm9m0u5092pu9ulq3chn0zkuks".to_string(),
+            query_data: serde_json::to_vec(&json!({"get_config": {}})).unwrap(),
+        };
+        let res = QuerySmartContractStateResponse {
+            data: serde_json::to_vec(&json!({
+                "name": "axelar-gateway",
+                "version": "1.0.0",
+                "owner": "axelar1q95p9fntvqn6jm9m0u5092pu9ulq3chn0zkuks",
+                "config": {
+                    "chain_id": "axelar-testnet-1",
+                    "enabled": true
+                }
+            }))
+            .unwrap(),
+        };
+
+        goldie::assert_json!(json!({
+            "req": format!("{:?}", req),
+            "res": format!("{:?}", res),
+            "req_bytes": req.to_bytes().unwrap(),
+            "res_bytes": res.to_bytes().unwrap()
+        }));
```

Comment:
what's the point of goldie here?

Reply:
Ensuring the request types we use don't change serialization when upgrading cosmrs version, and if they do, we catch it.